.gitlab-ci.yml

```diff
@@ -1,114 +1,138 @@
 
 test:native-ios:
-  tags: [ "macos:sonoma", "specific:true" ]
-  stage: test
-  timeout: 1h
-  script:
-    - yarn
-    - (cd example/ios && RCT_NEW_ARCH_ENABLED=0 pod install --repo-update)
-    - set -o pipefail && xcodebuild -workspace example/ios/DdSdkReactNativeExample.xcworkspace -scheme DatadogSDKReactNative test -destination "platform=iOS Simulator,OS=17.4,name=iPhone 15 Pro Max" | xcbeautify
+    tags: ['macos:sonoma', 'specific:true']
+    stage: test
+    rules:
+        - if: '$BUILD_BENCHMARK != "true"'
+    timeout: 1h
+    script:
+        - yarn
+        - (cd example/ios && RCT_NEW_ARCH_ENABLED=0 pod install --repo-update)
+        - set -o pipefail && xcodebuild -workspace example/ios/DdSdkReactNativeExample.xcworkspace -scheme DatadogSDKReactNative test -destination "platform=iOS Simulator,OS=17.4,name=iPhone 15 Pro Max" | xcbeautify
 
 test:native-ios-sr:
-  tags: [ "macos:sonoma", "specific:true" ]
-  stage: test
-  timeout: 1h
-  script:
-    - yarn
-    - (cd example/ios && RCT_NEW_ARCH_ENABLED=0 pod install --repo-update)
-    - set -o pipefail && xcodebuild -workspace example/ios/DdSdkReactNativeExample.xcworkspace -scheme DatadogSDKReactNativeSessionReplay test -destination "platform=iOS Simulator,OS=17.4,name=iPhone 15 Pro Max" | xcbeautify
+    tags: ['macos:sonoma', 'specific:true']
+    stage: test
+    rules:
+        - if: '$BUILD_BENCHMARK != "true"'
+    timeout: 1h
+    script:
+        - yarn
+        - (cd example/ios && RCT_NEW_ARCH_ENABLED=0 pod install --repo-update)
+        - set -o pipefail && xcodebuild -workspace example/ios/DdSdkReactNativeExample.xcworkspace -scheme DatadogSDKReactNativeSessionReplay test -destination "platform=iOS Simulator,OS=17.4,name=iPhone 15 Pro Max" | xcbeautify
 
 test:native-ios-newarch:
-  tags: [ "macos:sonoma", "specific:true" ]
-  stage: test
-  timeout: 1h
-  script:
-    - yarn
-    - (cd example-new-architecture/ios && RCT_NEW_ARCH_ENABLED=1 pod install --repo-update)
-    - set -o pipefail && xcodebuild -workspace example-new-architecture/ios/DdSdkReactNativeExample.xcworkspace -scheme DatadogSDKReactNative test -destination "platform=iOS Simulator,OS=17.4,name=iPhone 15 Pro Max" | xcbeautify
\ No newline at end of file
+    tags: ['macos:sonoma', 'specific:true']
+    stage: test
+    rules:
+        - if: '$BUILD_BENCHMARK != "true"'
+    timeout: 1h
+    script:
+        - yarn
+        - (cd example-new-architecture/ios && RCT_NEW_ARCH_ENABLED=1 pod install --repo-update)
+        - set -o pipefail && xcodebuild -workspace example-new-architecture/ios/DdSdkReactNativeExample.xcworkspace -scheme DatadogSDKReactNative test -destination "platform=iOS Simulator,OS=17.4,name=iPhone 15 Pro Max" | xcbeautify
+
+benchmark:
+    stage: benchmark
+    rules:
+        - if: '$BUILD_BENCHMARK == "true"'
+          allow_failure: true
+        - if: '$CI_COMMIT_BRANCH == "$DEVELOP_BRANCH"'
```

Comment:
Or when pushing to develop, meaning we need to update the benchmark apps to have the latest changes so we can run performance checks on them.